src/wovnio/wovnphp/Lang.php

```diff
@@ -1,116 +1,126 @@
+  }
 
+  /**
+   * Get the code of the lang given either a code, an original name or an English name\n
+   *
+   * @param String $lang_name Code, Original Name or English name of the lang
+   * @return String The code of the lang
+   */
+  public static function getCode($lang_name = null)
+  {
+    if ($lang_name === null) {
       return null;
     }
-
-    /**
-     * Get the code of the lang given either a code, an original name or an English name\n
-     *
-     * @param String $lang_name Code, Original Name or English name of the lang
-     * @return String The code of the lang
-     */
-    public static function getCode($lang_name=null) {
-      if ($lang_name === null) {
-        return null;
-      }
-      if (isset(Lang::$lang[$lang_name])) {
-        return $lang_name;
-      }
-      foreach(Lang::$lang as $lang) {
-        if (strtolower($lang_name) === strtolower($lang['name']) || strtolower($lang_name) === strtolower($lang['en']) || strtolower($lang_name) === strtolower($lang['code'])) {
-          return $lang['code'];
-        }
+    if (isset(Lang::$lang[$lang_name])) {
+      return $lang_name;
+    }
+    foreach (Lang::$lang as $lang) {
+      if (strtolower($lang_name) === strtolower($lang['name']) || strtolower($lang_name) === strtolower($lang['en']) || strtolower($lang_name) === strtolower($lang['code']) || strtolower($lang_name) === strtolower($lang['iso639-1'])) {
+        return $lang['code'];
       }
-      return null;
     }
+    return null;
+  }
 
-    /**
-     * Get the English names supported by WOVN++\n
-     *
-     * @return array The English names
-     */
-    public static function getEnglishNamesArray() {
-      $englishNamesArray = array();
-      foreach(Lang::$lang as $lang) {
-         array_push($englishNamesArray, $lang['en']);
-      }
-      return $englishNamesArray;
+  /**
+   * Get the English names supported by WOVN++\n
+   *
+   * @return array The English names
+   */
+  public static function getEnglishNamesArray()
+  {
+    $englishNamesArray = array();
+    foreach (Lang::$lang as $lang) {
+      array_push($englishNamesArray, $lang['en']);
     }
+    return $englishNamesArray;
+  }
 
-    /**
-     * Provides the ISO639-1 code for a given lang code.
-     * Source: https://support.google.com/webmasters/answer/189077?hl=en
-     *
-     * @param String $lang_code Code of the language.
-     *
-     * @return String The ISO639-1 code of the language.
-     */
-    public static function iso639_1Normalization($lang_code) {
-      return str_replace('zh-CHT', 'zh-Hant', str_replace('zh-CHS', 'zh-Hans', $lang_code));
+  /**
+   * Provides the ISO639-1 code for a given lang code.
+   * Source: https://support.google.com/webmasters/answer/189077?hl=en
+   *
+   * @param String $lang_code Code of the language.
+   *
+   * @return String The ISO639-1 code of the language.
+   */
+  public static function iso639_1Normalization($lang_code)
+  {
+    if (isset(Lang::$lang[$lang_code])) {
```

Comment:
I don't know if it is possible, but if user put custom lang alias in supported languages, wouldn't the custom lang alias be sent to this function?
If so, shouldn't we also resolve lang code with custom lang aliases before getting iso 639 code?

Reply:
we shouldn't use the custom lang alias for the href langs, only for the url

Reply:
Yes, I wan't a check here to convert custom lang code into wovn lang code for normalization.
@suplo, any plan to do that before merging?

Reply:
The previous version of this function didn't handle that case. From looking at the code, lang_code is always the wovn lang code, if it's custom it is specified.

I can modify to to support this case if you'd like. What do you think?

Reply:
Maybe ok from now, because previous version was like that.
I think we should make a task to either:
- support that case
- make sure user put wovn lang code in supported languages, not custom lang codes